### ISSUE TITLE
ECS TaskDefinition TaskRoleARN doesn't trigger replacement

### ISSUE THREAD

**Issue by xiongchiamiov** [pulumi/pulumi-aws-native#628]
### What happened?

I identified that an ECS service I had defined was not working because I had forgotten to add [an execution IAM role](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_execution_IAM_role.html).  Adding this to the Pulumi definition however produces an AWS API error.

### Steps to reproduce

1. Create an `ecs.TaskDefinition` with `pulumi up`
2. Add or change `task_role_arn`
3. Run `pulumi up` again

Pulumi code:
```python3
def cannot_change_arn():
    ecs.TaskDefinition(
        resource_name='cannot-change-arn',
        container_definitions=[
            ecs.TaskDefinitionContainerDefinitionArgs(
                name="my-app",
                image="amazon/amazon-ecs-sample",
                memory=128,
            )
        ],
        # After the first run to create the resource, uncomment this line to
        # try to update the object and reproduce the error.
        #task_role_arn='arn:aws:iam::1234567890:role/example',
    )
```
You can also clone https://github.com/xiongchiamiov/pulumi-bug-reports/tree/main/aws-native-python and call the function `ecs_task_definition.cannot_change_arn()` in `__main__.py`.

Pulumi run:
```
─[jamesph@Jamess-MBP] - [~/Documents/pulumi-bug-reports/aws-native-python] - [Mon Sep 19, 04:03]
└─[$]> pulumi up
Previewing update (dev)

View Live: https://app.pulumi.com/xiongchiamiov/aws-native-python/dev/previews/08a4536d-e9aa-44f9-afbc-c1368daa0198

     Type                              Name                   Plan
 +   pulumi:pulumi:Stack               aws-native-python-dev  create
 +   └─ aws-native:ecs:TaskDefinition  cannot-change-arn      create

Resources:
    + 2 to create

Do you want to perform this update?  [Use arrows to move, enter to select, type to filter
Do you want to perform this update? y  [Use arrows to move, enter to select, type to filt
Do you want to perform this update? ye  [Use arrows to move, enter to select, type to fil
Do you want to perform this update? yes  [Use arrows to move, enter to select, type to fi
Do you want to perform this update? yes
Updating (dev)

View Live: https://app.pulumi.com/xiongchiamiov/aws-native-python/dev/updates/12

     Type                              Name                   Status
 +   pulumi:pulumi:Stack               aws-native-python-dev  created
 +   └─ aws-native:ecs:TaskDefinition  cannot-change-arn      created

Resources:
    + 2 created

Duration: 4s
                                                                                                                                                                                   warning: A new version of Pulumi is available. To upgrade from version '3.39.3' to '3.40.1', run
   $ brew upgrade pulumi
┌─[jamesph@Jamess-MBP] - [~/Documents/pulumi-bug-reports/aws-native-python] - [Tue Sep 20, 12:45]
└─[$]> pulumi up
Previewing update (dev)

View Live: https://app.pulumi.com/xiongchiamiov/aws-native-python/dev/previews/ecd8dff9-543d-450d-b22f-149753affbf4

     Type                              Name                   Plan       Info
     pulumi:pulumi:Stack               aws-native-python-dev
 ~   └─ aws-native:ecs:TaskDefinition  cannot-change-arn      update     [diff: +taskRoleArn]

Resources:
    ~ 1 to update
    1 unchanged

Do you want to perform this update? details
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:dev::aws-native-python::pulumi:pulumi:Stack::aws-native-python-dev]
    ~ aws-native:ecs:TaskDefinition: (update)
        [id=arn:aws:ecs:us-west-2:755158074228:task-definition/ResjDQP9VXzzpqZ6DFIC2Suyw-CSnu4117INnc:1]
        [urn=urn:pulumi:dev::aws-native-python::aws-native:ecs:TaskDefinition::cannot-change-arn]
        [provider=urn:pulumi:dev::aws-native-python::pulumi:providers:aws-native::default_0_29_0::d949a3bc-0fa6-424b-8bdf-89818509bffc]
      + taskRoleArn: "arn:aws:iam::1234567890:role/example"

Do you want to perform this update? yes
Updating (dev)

View Live: https://app.pulumi.com/xiongchiamiov/aws-native-python/dev/updates/13

     Type                              Name                   Status                  Info
     pulumi:pulumi:Stack               aws-native-python-dev  **failed**              1 error
 ~   └─ aws-native:ecs:TaskDefinition  cannot-change-arn      **updating failed**     [diff: +taskRoleArn]; 1 error

Diagnostics:
  pulumi:pulumi:Stack (aws-native-python-dev):
    error: update failed

  aws-native:ecs:TaskDefinition (cannot-change-arn):
    error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: cfc5461e-218d-4934-8e67-98ba48fda343, NotUpdatableException: Invalid patch update: createOnlyProperties [/properties/TaskRoleArn] cannot be updated

Resources:
    1 unchanged

Duration: 2s

warning: A new version of Pulumi is available. To upgrade from version '3.39.3' to '3.40.1', run
   $ brew upgrade pulumi
or visit https://pulumi.com/docs/reference/install/ for manual instructions and release notes.
```

### Expected Behavior

I'd expect Pulumi to destroy the old object and create a new one, and update references to it.

### Actual Behavior

Crashes and that object is never able to be updated with Pulumi without doing a `pulumi destroy`/`pulumi up`.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by danielrbradley** [comment]
Thanks for filing this issue @xiongchiamiov.

Currently, the provider doesn't know about the nested property which should cause replacement due to the [upstream specifications](https://github.com/pulumi/pulumi-aws-native/blob/master/aws-cloudformation-schema/aws-ecs-taskdefinition.json) not yet indicating which fields require replacement. 

You should be able to workaround this issue by specifying the [`ReplaceOnChanges` resource option](https://www.pulumi.com/docs/intro/concepts/resources/options/replaceonchanges/) yourself. This should be something like:

```python
opts=ResourceOptions(replace_on_changes=["taskRoleArn"])
```

**Comment by danielrbradley** [comment]
Actually, on further reading of the specification, it might be possible to derive this behaviour, given that `/properties/TaskRoleArn` is listed in the `createOnlyProperties` set. We should probably therefore mark all of these properties as requiring replacement. This is probably dependent on #175

**Comment by 1oglop1** [comment]
@danielrbradley I found out that `containerDefinitions` do not respect the order and it is always different which gives a headache during diffs, is it because of this: https://github.com/pulumi/pulumi-aws-native/blob/e82ae3127a7e145f944c002cf7511bc51ccd3ba4/aws-cloudformation-schema/aws-ecs-taskdefinition.json#L736

Is this coming from the upstream or can that be simply changed to `true`?

PS. Is it worth creating a separate issue?

**Comment by xiongchiamiov** [comment]
Depending on the type of fix that gets put in, something that could be its own issue or part of the same one:  an error updating the cluster property of an `ecs.Service`:  `error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: 5a865bcc-55c6-4f6c-9a89-0811673fa793, NotUpdatableException: Invalid patch update: createOnlyProperties [/properties/Cluster] cannot be updated`

**Comment by mjeffryes** [comment]
This should be resolved by #1053; changes to `task_role_arn` will result in a replace rather than an update.
